sway/commands/reload.c

```diff
@@ -37,12 +37,26 @@ struct cmd_results *cmd_reload(int argc, char **argv) {
 		}
 	}
 
-	for (int i = 0; i < bar_ids->length; ++i) {
-		free(bar_ids->items[i]);
-	}
+	list_foreach(bar_ids, free);
 	list_free(bar_ids);
 
 	arrange_root();
+}
+
+struct cmd_results *cmd_reload(int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "reload", EXPECTED_EQUAL_TO, 0))) {
+		return error;
+	}
+
+	if (!load_main_config(config->current_config_path, true, true)) {
```

Comment:
This ostensibly works, but there is a bug when validating that seat configs are still being applied: it seems that they are not local to a config but global, so it still destroys the keyboard and the crash I was trying to fix before still occurs.

Reply:
Can you give some example config that produces the crash?

Reply:
Actually, the problem is more subtle than I made out. It seems that if you have any seat configuration before setting a fallback, like:
```
seat s attach "fake_input"
seat seat0 fallback true
```

instead of the other way round, the first command causes all the devices to be removed from the fallback seat, destroying them, before the next command reattaches them.

My full minimal config is:
```
input "1:1:AT_Translated_Set_2_keyboard" {
    xkb_layout us
    xkb_variant dvorak
}

seat s attach "fake_input"
seat seat0 fallback true

bindsym Mod4+Shift+c reload
bindsym Mod4+Return exec termite
```
but swapping the two aforementioned lines prevents the crash.

Reply:
Should be resolved now.